packages/calcite-components/src/components/list/list.e2e.ts

```diff
@@ -136,153 +159,259 @@ describe("calcite-list", () => {
+  it("should set the mode property on items", async () => {
+    const page = await newE2EPage();
+    await page.setContent(
+      html`<calcite-list id="root" mode="nested" group="my-list">
+        <calcite-list-item open label="Depth 1" description="Item 1">
+          <calcite-list group="my-list">
+            <calcite-list-item open label="Depth 2" description="Item 2">
+              <calcite-list mode="nested" group="my-list">
+                <calcite-list-item label="Depth 3" description="Item 3">
+                  <calcite-list mode="nested" group="my-list"></calcite-list>
+                </calcite-list-item>
+                <calcite-list-item label="Depth 3" description="Item 4"></calcite-list-item>
+              </calcite-list>
+            </calcite-list-item>
+            <calcite-list-item label="Depth 2" description="Item 5"></calcite-list-item>
+          </calcite-list>
+        </calcite-list-item>
+        <calcite-list-item label="Depth 1" description="Item 6"></calcite-list-item>
+        <calcite-list-item drag-disabled label="Depth 1" description="Item 7"></calcite-list-item>
+      </calcite-list>`,
+    );
+
+    await page.waitForChanges();
+    await page.waitForTimeout(DEBOUNCE.filter);
+
+    let modeValues = ["nested", "flat", "nested", "nested", "flat", "nested", "nested"];
+
+    const items = await page.findAll("calcite-list-item");
+
+    expect(items.length).toBe(modeValues.length);
+
+    for (let i = 0; i < items.length; i++) {
+      expect(await items[i].getProperty("mode")).toBe(modeValues[i]);
+    }
+
+    const rootList = await page.find("#root");
+
+    rootList.setProperty("mode", "flat");
+    await page.waitForChanges();
+    await page.waitForTimeout(DEBOUNCE.filter);
+
+    modeValues = ["flat", "flat", "nested", "nested", "flat", "flat", "flat"];
+
+    expect(items.length).toBe(modeValues.length);
+
+    for (let i = 0; i < items.length; i++) {
+      expect(await items[i].getProperty("mode")).toBe(modeValues[i]);
+    }
+  });
+
+  it("should set the mode property on items which are not direct children", async () => {
+    const page = await newE2EPage();
+    await page.setContent(
+      html`<calcite-list id="root" mode="nested" group="my-list">
+        <div>
           <calcite-list-item open label="Depth 1" description="Item 1">
             <calcite-list group="my-list">
-              <calcite-list-item open label="Depth 2" description="Item 2">
-                <calcite-list drag-enabled group="my-list">
-                  <calcite-list-item label="Depth 3" description="Item 3">
-                    <calcite-list drag-enabled group="my-list"></calcite-list>
-                  </calcite-list-item>
-                  <calcite-list-item label="Depth 3" description="Item 4"></calcite-list-item>
-                </calcite-list>
-              </calcite-list-item>
-              <calcite-list-item label="Depth 2" description="Item 5"></calcite-list-item>
+              <div>
```

Comment:
What's the use case for wrapped list items? If it's not a strong one, I'd suggest not supporting this to keep things simple. This also aligns with the default slot's description where we only expect items and groups. Also applies to similar `dragHandle` test.

Reply:
This is basically a copy from an existing test for dragHandle. I can remove the test though.

Reply:
Yeah, let's remove.